rls/src/actions/mod.rs

```diff
@@ -425,6 +435,24 @@ impl InitActionContext {
     }
 }
 
+/// Read edition from the manifest
+fn edition_from_manifest<P: AsRef<Path>>(manifest_path: P) -> Option<Edition> {
```

Comment:
I'm not thrilled about placement of this function here but maybe it's self-contained enough that it's not a problem.

Reply:
It's a private function that's only used here so it seems ok to me. I'd imagine it won't be hard to move or replace later. But if you can think of a better place to put it, along with the test, then sure.